src/pymor/algorithms/newton.py

```diff
@@ -4,14 +4,16 @@
 
 import numpy as np
 
+from pymor.algorithms.line_search import armijo
+
 from pymor.core.defaults import defaults
 from pymor.core.exceptions import InversionError, NewtonError
 from pymor.core.logger import getLogger
 
 
 @defaults('miniter', 'maxiter', 'rtol', 'atol', 'relax', 'stagnation_window', 'stagnation_threshold')
 def newton(operator, rhs, initial_guess=None, mu=None, error_norm=None, least_squares=False,
-           miniter=0, maxiter=100, rtol=-1., atol=-1., relax=1.,
+           miniter=0, maxiter=100, rtol=-1., atol=-1., relax=1., line_search_method='armijo',
```

Comment:
Does it make sense to set the default to 'armijo' ? Is it always recommended to go for a line_search_method here? If not, I would go for None as default.

Reply:
Certainly, we can set `None` as default. The advantage in that case would be that the user immediately notices if the Newton method needs to be adjusted. Then, he can decide whether to try a different value for `relax` (which can be understood as a line search with fixed outcome) or to use a line search method. Using `armijo` or another line search method as default would mean that the user does not see any adjustments the line search method makes.

Reply:
I agree

Reply:
Good idea, that sounds reasonable.

Reply:
I changed this part in a3f6844. I left `relax=1.` as default.